srml/contract/src/rent.rs

```diff
@@ -0,0 +1,138 @@
+use crate::{BalanceOf, ContractInfo, ContractInfoOf, Module, TombstoneContractInfo, Trait};
+use runtime_primitives::traits::{As, Bounded, CheckedDiv, CheckedMul, Saturating, Zero};
+use srml_support::traits::{Currency, ExistenceRequirement, Imbalance, WithdrawReason};
+use srml_support::StorageMap;
+
+/// Evict and optionnaly pay rent, at block number. Return if evicted
+///
+/// Exempted from rent iff:
+/// * rent is offset completely by the `rent_deposit_offset`,
+/// * or rent has already been paid for this block number,
+/// * or account doesn't have a contract,
+/// * or account has a tombstone.
+///
+/// Evicted iff:
+/// * rent exceed rent allowance,
+/// * or can't withdraw the rent,
+/// * or go below subsistence threshold.
+fn try_evict_or_and_pay_rent<T: Trait>(
+	account: &T::AccountId,
+	block_number: T::BlockNumber,
+	pay_rent: bool,
+) -> bool {
```

Comment:
I wonder if it is worth to introduce an enum instead of this `bool`. This will help us document exactly what `bool` in `pay_rent` or `try_evict_at` means (not it is not clear immediately IMO)